File: ios/OutdoorApp/Sources/Models/Activity.swift
import Foundation

struct Activity: Codable {
    let id: Int?
    var name: String
    var type: String?
    var location: String?
    var startDate: String?
    var duration: String?
    var description: String?
    var totalDistance: Double?
    var totalElevation: Int?
    var coverIcon: String?
    var userId: Int?
    var favorite: Bool?
    var createdAt: String?
    var updatedAt: String?

    enum CodingKeys: String, CodingKey {
        case id, name, type, location, duration, description
        case totalDistance = "total_distance"
        case totalElevation = "total_elevation"
        case coverIcon = "cover_icon"
        case userId = "user_id"
        case favorite
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case startDate = "start_date"
    }
}

struct ActivityCreateReq: Codable {
    let name: String
    let type: String?
    let location: String?
    let startDate: String?
    let duration: String?
    let description: String?

    enum CodingKeys: String, CodingKey {
        case name, type, location, duration, description
        case startDate = "start_date"
    }
}

File: src/main/java/com/ruiyu/outdoor/model/Activity.java
package com.ruiyu.outdoor.model;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 户外活动实体
 */
@Data
@TableName("activities")
@AllArgsConstructor
@NoArgsConstructor
public class Activity {

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 活动名称 */
    private String name;

    /** 活动类型: hiking-徒步, camping-露营, climbing-登山, cycling-骑行 */
    private String type;

    /** 活动地点 */
    private String location;

    /** 开始日期 */
    private LocalDate startDate;

    /** 预计时长 */
    private String duration;

    /** 活动描述 */
    private String description;

    /** 总距离(公里) */
    private Double totalDistance;

    /** 总爬升(米) */
    private Integer totalElevation;

    /** 封面图标 */
    private String coverIcon;

    /** 用户ID */
    private Integer userId;

    /** 是否收藏 */
    private Boolean favorite;

    /** 创建时间 */
    private LocalDateTime createdAt;

    /** 更新时间 */
    private LocalDateTime updatedAt;
}
